pkg/app/pipedv1/deploysource/deploysource.go

```diff
@@ -191,6 +191,13 @@ func (p *provider) copy(lw io.Writer) (*DeploySource, error) {
 		return nil, err
 	}
 
+	// To avoid the git operation on copied source.
+	// TODO: do not copy the .git directory to improve the performance.
+	if err := os.RemoveAll(filepath.Join(dest, ".git")); err != nil {
+		fmt.Fprintf(lw, "Unable to remove the .git directory from the copied deploy source (%v)\n", err)
+		return nil, err
+	}
```

Comment:
How about using a `tar` trick to avoid `.git` copy and remove. I think tar is available everywhere and fast enough compared to cp
```
tar -cpP --exclude=.git -C "$source_dir" . | tar -xpP -C "$target_dir"
```

Reply:
Thanks. I'll try.

Reply:
@khanhtc1202 

We use the alpine as the base image for piped, so I checked the tar command in the piped image.
First, the tar command exists. But, the options differ from the GNU/BSD tar.
For example, short flags `-p` and `-P` do not exist.
IMO, it's not good to depend on this tar behavior because users may run the piped on a non-alpine-based OS.
WDYT?

Note: I checked `tar --help` in the piped container image. I paste the result below.
```
$ docker run --rm -it --entrypoint sh  ghcr.io/pipe-cd/piped:v0.50.0-23-g8f23da4@sha256:88ec60ebf2b7e3131ff38b1af4a043117ef688bc636ddd95afd6bd368293651b
/ $ tar --help
BusyBox v1.36.1 (2024-06-10 07:11:47 UTC) multi-call binary.

Usage: tar c|x|t [-ZzJjahmvokO] [-f TARFILE] [-C DIR] [-T FILE] [-X FILE] [LONGOPT]... [FILE]...

Create, extract, or list files from a tar file

        c       Create
        x       Extract
        t       List
        -f FILE Name of TARFILE ('-' for stdin/out)
        -C DIR  Change to DIR before operation
        -v      Verbose
        -O      Extract to stdout
        -m      Don't restore mtime
        -o      Don't restore user:group
        -k      Don't replace existing files
        -Z      (De)compress using compress
        -z      (De)compress using gzip
        -J      (De)compress using xz
        -j      (De)compress using bzip2
        --lzma  (De)compress using lzma
        -a      (De)compress based on extension
        -h      Follow symlinks
        -T FILE File with names to include
        -X FILE File with glob patterns to exclude
        --exclude PATTERN       Glob pattern to exclude
        --overwrite             Replace existing files
        --strip-components NUM  NUM of leading components to strip
        --no-recursion          Don't descend in directories
        --numeric-owner         Use numeric user:group
        --no-same-permissions   Don't restore access permissions
```

Reply:
This is the version without -P, the more universal solution
```
tar --exclude='.git' -cf - source_dir | tar -xf - -C destination_dir
```
@Warashi

Reply:
OKay, thank you for providing this. I'll use this script.

Reply:
I tried some tar options and finally reached this commit.
I tested it with busybox's tar (the same as alpine's) and GNU tar.
https://github.com/pipe-cd/pipecd/pull/5412/commits/c5216c91bffc906f57b30133f3d0e557f1b1888f